scripts/install/providers/azdo/azdo.sh

```diff
@@ -133,6 +167,133 @@ function configure_repo {
+  RUNNERS_AZURE_VM_SIZE=${RUNNERS_AZURE_VM_SIZE:-Standard_D2s_v3}
+
+  # Optional parameter: Number of runners to launch (default: 1)
+  RUNNERS_COUNT=${RUNNERS_COUNT:-1}
+
+  # Optional parameter: Image (default: Ubuntu2404)
+  RUNNERS_AZURE_IMAGE=${RUNNERS_AZURE_IMAGE:-Ubuntu2404}
+
+  # Optional parameter: VM name (default: symphony-azdo-runner)
+  RUNNERS_VM_NAME=${RUNNERS_VM_NAME:-symphony-azdo-runner}-${AZDO_PROJECT_NAME}
+
+  # Optional parameter: VM username (default: azureuser)
+  RUNNERS_VM_USERNAME=${RUNNERS_VM_USERNAME:-azureuser}
+
+  local token=$AZDO_PAT
+
+  # Generate the cloud-init file (cloud-init.yaml)
+  cat > cloud-init.yaml <<EOF
+  #cloud-config
+  package_update: true
+
+  packages:
+    - curl
+    - tar
+    - uidmap
+    - unzip
+    - docker.io
+    - build-essential
+
+  write_files:
+    - path: /agent-install.sh
+      permissions: '0777'
+      content: |
+          #!/usr/bin/env bash
+
+          cd /home/$RUNNERS_VM_USERNAME
+
+          # Enable Docker
+          sudo systemctl enable --now docker
+          sudo usermod -aG docker $RUNNERS_VM_USERNAME
+
+          # Install AzDO runner
+
+          for i in \$(seq 1 ${RUNNERS_COUNT}); do
+              RUNNER_DIR="/home/$RUNNERS_VM_USERNAME/azdo-runner-\$i"
+              echo "Configuring runner in \$RUNNER_DIR..."
+              mkdir -p "\$RUNNER_DIR"
+              cd "\$RUNNER_DIR"
+
+              # Download a fixed version of the runner
+              RUNNER_VERSION="4.251.0"
+              curl -O -L "https://vstsagentpackage.azureedge.net/agent/\${RUNNER_VERSION}/vsts-agent-linux-x64-\${RUNNER_VERSION}.tar.gz"
+              tar xzf "vsts-agent-linux-x64-\${RUNNER_VERSION}.tar.gz"
+
+              # Configure the runner non-interactively using its respective token
+              ./config.sh --unattended --url "${AZDO_ORG_URI}" --auth pat --token "${token}" --pool "Default" --agent "agent-\$i" --acceptTeeEula
+
+              # Install the service
+              sudo ./svc.sh install
+
+              # Start the runner in the background
+              sudo ./svc.sh start
+
+              # Return to the home directory for the next iteration
+              cd /home/$RUNNERS_VM_USERNAME
+          done
+  runcmd:
+    - sudo -u $RUNNERS_VM_USERNAME /agent-install.sh
+EOF
+
+  echo "cloud-init.yaml file generated."
+
+  # Create the VM using Azure CLI with the specified parameters and cloud-init
+  _information "Creating the VM in Azure..."
+  vm_create_command="az vm create \
+    --resource-group \"$RUNNERS_RESOURCE_GROUP\" \
+    --name \"$RUNNERS_VM_NAME\" \
+    --location \"$RUNNERS_LOCATION\" \
+    --image \"$RUNNERS_AZURE_IMAGE\" \
+    --size \"$RUNNERS_AZURE_VM_SIZE\" \
+    --admin-username \"$RUNNERS_VM_USERNAME\" \
+    --ssh-key-values \"@$RUNNERS_PUBLIC_KEY_PATH\" \
+    --subnet \"$RUNNERS_AZURE_SUBNET\" \
+    --nsg \"\" \
+    --custom-data cloud-init.yaml"
+
+  eval $vm_create_command
```

Comment:
nit: consider adding an optional debug output for vm_create_command so users who run this with the -d option can see the command.